### DISCUSSION TITLE
Flask Socket IO freeze in between

### DISCUSSION THREAD

**Discussion by kartikshetty0897** [miguelgrinberg/Flask-SocketIO#1849] [Q&A]
I am building a real time image detection webapp using angular, flask and socketio. I am using socketio to send frames of images form the client to the server. After a few requests the socketio application stops running and suddenly starts working discarding all the requests between the freeze and restart. And some times the socketio server starts after some time of running the application. Can someone please help me in identifying what might the issue be ?

**Comment by miguelgrinberg** [comment]
Are you using an async server (gevent, eventlet)? Are you familiar with these frameworks and how to work with them to prevent blocking?

**Reply by kartikshetty0897** [reply]
I am using eventlet to monkey patch and the flask server

**Reply by miguelgrinberg** [reply]
Monkey patching helps with some tasks, but you could still have blocking code in your application that isn't supported by monkey patching. I suggest you retest using `threading` mode without eventlet, to see if you still get the blocking. At least that will give you an idea of wether this is due to eventlet blocking or something more general.
